### PR TITLE
Switch sample apiserver internalversion informer to external

### LINKED ISSUE #71788: sample apiserver report failed to list *wardle.Fischer: the server could not find the requested resource
**What happened**:
sample api server log:
```
I1206 19:53:40.714834   63361 round_trippers.go:405] GET https://[::1]:8443/apis/wardle.k8s.io/v1beta1/fischers?limit=500&resourceVersion=0 404 Not Found in 1 milliseconds
I1206 19:53:40.714855   63361 round_trippers.go:411] Response Headers:
I1206 19:53:40.714864   63361 round_trippers.go:414]     Content-Type: application/json
I1206 19:53:40.714870   63361 round_trippers.go:414]     Content-Length: 174
I1206 19:53:40.714876   63361 round_trippers.go:414]     Date: Thu, 06 Dec 2018 11:53:40 GMT
I1206 19:53:40.714906   63361 request.go:942] Response Body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"the server could not find the requested resource","reason":"NotFound","details":{},"code":404}
E1206 19:53:40.715003   63361 reflector.go:134] k8s.io/sample-apiserver/pkg/client/informers/internalversion/factory.go:117: Failed to list *wardle.Fischer: the server could not find the requested resource
```
registry:
```
	v1alpha1storage["flunders"] = wardleregistry.RESTInPeace(flunderstorage.NewREST(Scheme, c.GenericConfig.RESTOptionsGetter))
	v1alpha1storage["fischers"] = wardleregistry.RESTInPeace(fischerstorage.NewREST(Scheme, c.GenericConfig.RESTOptionsGetter))
	apiGroupInfo.VersionedResourcesStorageMap["v1alpha1"] = v1alpha1storage

	v1beta1storage := map[string]rest.Storage{}
	v1beta1storage["flunders"] = wardleregistry.RESTInPeace(flunderstorage.NewREST(Scheme, c.GenericConfig.RESTOptionsGetter))
	apiGroupInfo.VersionedResourcesStorageMap["v1beta1"] = v1beta1storage
``` 

It seems that the internalversion's informer list watch v1beta1 fischers API, but the fischers only have the v1alpha1 API in sample apiserver. where determine which version of the API the internalversion informer calls? another questions is ```Admit(a admission.Attributes)```  access attributes，can we directly use versioned informer？

/sig api-apimachinery

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/kind bug

### PR DESCRIPTION
**Pull request by zjj2wry**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/71788

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by liggitt** [comment]
beta should still be preferred over alpha. the better solution is to stop using the internal fischer informer and use a versioned one specifically pointed at the v1alpha1 API. this aligns with the work in progress to stop using internal informers for built-in types as well (https://github.com/kubernetes/kubernetes/issues/70657)

**Comment by zjj2wry** [comment]
thanks, i will update this PR  to fix it

File: staging/src/k8s.io/sample-apiserver/hack/update-codegen.sh [hunk 1]
```diff
@@ -25,8 +25,8 @@ CODEGEN_PKG=${CODEGEN_PKG:-$(cd ${SCRIPT_ROOT}; ls -d -1 ./vendor/k8s.io/code-ge
 # --output-base    because this script should also be able to run inside the vendor dir of
 #                  k8s.io/kubernetes. The output-base is needed for the generators to output into the vendor dir
 #                  instead of the $GOPATH directly. For normal projects this can be dropped.
-${CODEGEN_PKG}/generate-internal-groups.sh all \
-  k8s.io/sample-apiserver/pkg/client k8s.io/sample-apiserver/pkg/apis k8s.io/sample-apiserver/pkg/apis \
+${CODEGEN_PKG}/generate-groups.sh all \
+  k8s.io/sample-apiserver/pkg/client k8s.io/sample-apiserver/pkg/apis \
```
Comment (liggitt) [inline staging/src/k8s.io/sample-apiserver/hack/update-codegen.sh]: cc @sttts

File: staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission.go [hunk 2]
```diff
@@ -85,11 +85,11 @@ func (d *DisallowFlunder) Admit(a admission.Attributes) error {
...
 // The lister knows how to lists Fischers.
 func (d *DisallowFlunder) SetInternalWardleInformerFactory(f informers.SharedInformerFactory) {
-	d.lister = f.Wardle().InternalVersion().Fischers().Lister()
-	d.SetReadyFunc(f.Wardle().InternalVersion().Fischers().Informer().HasSynced)
+	d.lister = f.Wardle().V1alpha1().Fischers().Lister()
+	d.SetReadyFunc(f.Wardle().V1alpha1().Fischers().Informer().HasSynced)
 }
 
-// ValidaValidateInitializationte checks whether the plugin was correctly initialized.
+// ValidaValidateInitialization checks whether the plugin was correctly initialized.
```
Comment (liggitt) [inline staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission.go]: preexisting, but while we're in here: ValidateInitialization

File: staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission_test.go [hunk 3]
```diff
@@ -56,7 +56,7 @@ func TestBanflunderAdmissionPlugin(t *testing.T) {
 					Namespace: "",
 				},
 			},
-			admissionInputKind:     wardle.Kind("Flunder").WithVersion("version"),
```
Comment (liggitt) [inline staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission_test.go]: these changes shouldn't be necessary

**Comment by liggitt** [comment]
a couple nits, would like @sttts to ack the generate-groups.sh invocation, lgtm otherwise

File: staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission_test.go (hunk 3 above)
Comment (zjj2wry) [reply staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission_test.go]: when using the internal API structure to execute the test will panic
```
/usr/local/go/src/runtime/asm_amd64.s:1333
panic: interface conversion: runtime.Object is *wardle.FischerList, not *v1alpha1.FischerList [recovered]
	panic: interface conversion: runtime.Object is *wardle.FischerList, not *v1alpha1.FischerList

goroutine 26 [running]:
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/Users/admin/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x108
panic(0x1b960c0, 0xc0001c34a0)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
k8s.io/kubernetes/vendor/k8s.io/sample-apiserver/pkg/client/clientset/versioned/typed/wardle/v1alpha1/fake.(*FakeFischers).List(0xc0000c0398, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/admin/go/src/k8s.io/kubernetes/vendor/k8s.io/sample-apiserver/pkg/client/clientset/versioned/typed/wardle/v1alpha1/fake/fake_fischer.go:62 +0x5f9
k8s.io/kubernetes/vendor/k8s.io/sample-apiserver/pkg/client/informers/externalversions/wardle/v1alpha1.NewFilteredFischerInformer.func1(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1ca3ef7, ...)
	/Users/admin/go/src/k8s.io/kubernetes/vendor/k8s.io/sample-apiserver/pkg/client/informers/externalversions/wardle/v1alpha1/fischer.go:63 +0x126
```

File: staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission.go (hunk 2 above)
Comment (zjj2wry) [reply staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission.go]: done

File: staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission_test.go (hunk 3 above)
Comment (liggitt) [reply staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission_test.go]: I was referring specifically to the groupversionkind produced by this line... the old and new lines should have identical output

Comment (zjj2wry) [reply staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder/admission_test.go]: No Kind() method in versioned/register.go. so i add this line

**Comment by zjj2wry** [comment]
@sttts @liggitt PTAL

File: staging/src/k8s.io/sample-apiserver/hack/update-codegen.sh (hunk 1 above)
Comment (sttts) [reply staging/src/k8s.io/sample-apiserver/hack/update-codegen.sh]: E.g. we could move internal generation to "{client,informer,lister}-internal" and remove them from the "all" logic. This way there is a way to generate the internal variants for those who need them, but they have to add them manually. Seems to be fair.

Comment (zjj2wry) [reply staging/src/k8s.io/sample-apiserver/hack/update-codegen.sh]: thanks for review, I added generate-internal-groups back, but only generate defaulter, deep copy and conversion

**Comment by zjj2wry** [comment]
@sttts @liggitt PTAL

**Comment by zjj2wry** [comment]
/retest
@sttts @liggitt PTAL

**Comment by zjj2wry** [comment]
kindly ping @sttts @liggitt PTAL

**Pull request merged**